functions/src/update-metrics.ts

```diff
@@ -133,6 +133,41 @@ export const updateMetricsCore = async () => {
     'set'
   )
   log(`Updated metrics for ${users.length} users.`)
+
+  const badgesUpdates = await userUpdatesWithBadges(users)
+  await writeAsync(firestore, badgesUpdates)
+}
+
+const userUpdatesWithBadges = async (users: User[]) => {
+  const [topTenTraders, topTenCreators] = await Promise.all([
+    getValues<User>(
+      firestore.collection('users').orderBy('profitCached', 'desc').limit(10)
+    ),
+    getValues<User>(
+      firestore
+        .collection('users')
+        .orderBy('creatorVolumeCached', 'desc')
+        .limit(10)
+    ),
+  ])
+
+  const userUpdatesWithBadges = users.map((user) => {
```

Comment:
This updates every user, so it does reset the ones who lost their rank. IndexOf returns -1 if the user isn't in topTenCreators. This means that it'll store 1-10 for the current top10 creators, and 0 for everybody else. 

I don't love updating every user when it could just be the current top 10 and the previous top 10, but these won't be the only badges, and all of them will involve updating separate sets of users each time the function runs (Ex: markets created streak in days, >100 markets created, top bettor in x community, 1th/10th/100th bet made, etc).